src/components/ElevationGraph.vue

```diff
@@ -0,0 +1,442 @@
+            .attr('x1', 0)
+            .attr('y1', marginTop)
+            .attr('x2', 0)
+            .attr('y2', height - marginBottom)
+            .attr('stroke', '#78e08f')
+            .attr('stroke-width', 1.5)
+        })
+        .call((g) => {
+          g.append('polygon').attr('points', '0,30 -5,20 5,20').attr('fill', '#78e08f')
+        })
+
+      // Add Events
+      const pointerListener = (e: Event) => {
+        const [px] = d3.pointer(e)
+        const [xMin, xMax] = xScale.range()
+
+        pointer.style('opacity', 1)
+
+        if (px <= xMin) {
+          pointer.attr('transform', `translate(${xMin}, 0)`)
+          this.hoveredRecord = this.records[0]
+          return
+        } else if (px >= xMax) {
+          pointer.attr('transform', `translate(${xMax}, 0)`)
+          this.hoveredRecord = this.records[this.records.length - 1]
+          return
+        }
+
+        pointer.attr('transform', `translate(${px}, 0)`)
+
+        const pointerPercentage = (px - xMin) / (xMax - xMin)
+        const lookupIndex = Math.round(pointerPercentage * this.records.length - 1)
+
+        let nearestRecord: Record = new Record()
+        let dx = Number.MAX_VALUE
+        // let counter = 0 // TODO: remove after debug (currently it's the fastest lookup)
+        if (xScale(this.records[lookupIndex].distance! / 1000) <= px) {
+          for (let i = lookupIndex; i < this.records.length; i++) {
+            const delta = Math.abs(px - xScale(this.records[i].distance! / 1000)!)
+            if (delta > dx) break
+            nearestRecord = this.records[i]
+            dx = delta
+            // counter++
+          }
+          //   console.debug(`look forward for ${counter} records`)
+        } else {
+          for (let i = lookupIndex; i >= 0; i--) {
+            const delta = Math.abs(px - xScale(this.records[i].distance! / 1000)!)
+            if (delta > dx) break
+            nearestRecord = this.records[i]
+            dx = delta
+            // counter++
+          }
+          //   console.debug(`look backward for ${counter} records`)
+        }
+
+        this.hoveredRecord = nearestRecord
+      }
+
+      svg.on('pointerdown', pointerListener)
+      svg.on('pointermove', pointerListener, { passive: true })
+      svg.on('mouseleave', () => {
+        this.hoveredRecord = new Record()
+        pointer.style('opacity', 0)
+      })
+    },
+    onResize() {
+      // Ensure DOM is fully re-rendered after resize
+      this.$nextTick(() => {
+        // Prevent re-render graph when width is not changing
+        if (window.innerWidth == this.previousWindowWidth) return
+        this.previousWindowWidth = window.innerWidth
+        this.renderGraph()
+      })
+    }
+  },
+  mounted() {
+    this.$nextTick(() => (this.previousWindowWidth = window.innerHeight))
+    window.addEventListener('resize', this.onResize)
+  },
+  unmounted() {
+    window.removeEventListener('resize', this.onResize)
+  }
+}
+</script>
+
+<style>
```

Comment:
This style can affect globally. please use scope if intended only for this component.